### PR TITLE
RSP-1710: Input validation

### PR DESCRIPTION
**Pull request by jameshighfieldbjss**
Update validation to restrict the use of special characters (`&, <, >, ‘, and ”`). Note that `'` is different from `‘`. Bump version to `1.5.16`.

### CODE REVIEW

File: src/ValidationModels/penaltyValidation.js [hunk 1]
```diff
@@ -1,14 +1,16 @@
 import Joi from 'joi';
 
+const regNoValidation = Joi.string().min(1).max(10)
```
Comment (johnmorleybjss) [inline src/ValidationModels/penaltyValidation.js]: Does the app allow input of spaces for reg no? Or does it strip them out before submission?

File: src/ValidationModels/penaltyValidation.js [hunk 2]
```diff
@@ -21,14 +23,14 @@ const trailerSchema = {
 
 const valueSchema = {
 	penaltyType: Joi.string().regex(/^(IM|CDN|FPN)$/).required(),
-	paymentStatus: Joi.string(), // .regex(/^(UNPAID|PAID)$/).required(),
-	paymentAuthCode: Joi.string(),
+	paymentStatus: Joi.string().valid(['PAID', 'UNPAID']),
```
Comment (johnmorleybjss) [inline src/ValidationModels/penaltyValidation.js]: I see `.required` is gone here, is that intentional?

File: src/ValidationModels/penaltyValidation.js [hunk 3]
```diff
@@ -21,14 +23,14 @@ const trailerSchema = {
...
+	paymentAuthCode: Joi.string().alphanum(),
 	// .when('paymentStatus', { is: 'PAID', then: Joi.required() }),
 	paymentDate: Joi.number().integer(),
 	// .when('paymentStatus', { is: 'PAID', then: Joi.required() }),
-	paymentToken: Joi.string().required(),
-	formNo: Joi.string(),
```
Comment (johnmorleybjss) [inline src/ValidationModels/penaltyValidation.js]: `formNo` is gone, was it unused?

File: src/ValidationModels/penaltyValidation.js [hunk 4]
```diff
@@ -40,19 +42,19 @@ const valueSchema = {
...
-		ID: Joi.string().required(),
-		Hash: Joi.string().required(),
+		ID: Joi.string().token().required(),
+		Hash: Joi.string().required().alphanum(),
 		Enabled: Joi.boolean().required(),
 		Offset: Joi.number(),
 		Value: Joi.object(valueSchema),
-		Origin: Joi.string(),
-		VehicleRegistration: Joi.string(),
+		Origin: Joi.string().valid(['APP']),
```
Comment (johnmorleybjss) [inline src/ValidationModels/penaltyValidation.js]: I feel like individual penalty documents can have an `Origin` other than `APP` somehow. Probably worth some investigation, the purpose this field has always been a bit beyond me.

Comment (jameshighfieldbjss) [reply src/ValidationModels/penaltyValidation.js]: I couldn't see any example of the Origin being anything else unless it was manually set during testing. I'll change it to allow A-Z to be safe.

File: src/ValidationModels/penaltyValidation.js (hunk 3 above)
Comment (jameshighfieldbjss) [reply src/ValidationModels/penaltyValidation.js]: Yes it's never assigned a value in the iOS app and none of the services use it.

File: src/ValidationModels/penaltyValidation.js (hunk 2 above)
Comment (jameshighfieldbjss) [reply src/ValidationModels/penaltyValidation.js]: Yes, `.required` was commented out and the app uploads documents without a payment status.

File: src/ValidationModels/penaltyValidation.js (hunk 1 above)
Comment (jameshighfieldbjss) [reply src/ValidationModels/penaltyValidation.js]: It strips them out. We previously had that validation only within `vehicleSchema` but not for `VehicleRegistration`.

**Pull request merged**